packages/apps/auth0-actions/src/login.js

```diff
@@ -36,7 +36,7 @@ function login(email, password, callback) {
           '/v6/patrons/validate',
           {
             barcode: barcode,
-            pin: pin,
+            pin: encodeURI(pin),
```

Comment:
Will this work? See "not escaped" list here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURI#description

Reply:
`encodeURIComponent` will handle more, but not the `!` - what encoding did you find worked for that? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent#description

Reply:
FWIW I didn't have a problem with `!` when I was testing this directly against the Sierra API.

Reply:
> FWIW I didn't have a problem with ! when I was testing this directly against the Sierra API.
Indeed testing this in isolation (in a different password from what I was using before) it does seem to be fine with ! not encoded.

Reply:
Testing makes it look like we need encodeURIComponent here.